src/main/clojure/clara/rules/engine.cljc

```diff
@@ -937,15 +950,19 @@
      memory
      listener
      children
-     (filter test tokens)))
+     (filter (partial test-node-matches node test {}) tokens)))
```

Comment:
FYI: `partial` degrades performance possibly significantly beyond it's 4 arity case. So this is pushing it right up against the limit.

Reply:
Good point, I'm going to remove the `env` argument since it doesn't really get used, I was just sticking to the existing pattern. Another possibility is to write this as an eager-for comprehension, so I'm going to give that a try.